### PR TITLE
enh(auth): Handle groups mapping on openId Authentication

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to Handle groups mapping on openId Authentication

**Fixes** # MON-15088

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 1]
```diff
@@ -1235,4 +1239,122 @@ public function getRolesMappingFromProvider(): array
     {
         return $this->rolesMappingFromProvider;
     }
+
+    /**
+     * @param GroupsMapping $groupsMapping
+     */
+    private function validateGroupsMappingOrFail(GroupsMapping $groupsMapping): void {
```
Comment (kduret) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
    private function validateGroupsMappingOrFail(GroupsMapping $groupsMapping): void
    {
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 2]
```diff
@@ -1235,4 +1239,122 @@ public function getRolesMappingFromProvider(): array
...
+        } else {
+            $this->logAuthenticationInfo("Groups Mapping disabled", []);
+        }
+    }
+
+    /**
+     * @param Endpoint $endpoint
+     * @return array<string, mixed>
+     */
+    private function getGroupsFromProvider(Endpoint $endpoint): array {
```
Comment (kduret) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
    private function getGroupsFromProvider(Endpoint $endpoint): array
    {
```

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 3]
```diff
@@ -1235,4 +1239,122 @@ public function getRolesMappingFromProvider(): array
...
+
+    /**
+     * @param GroupsMapping $groupsMapping
+     */
+    private function validateGroupsMappingOrFail(GroupsMapping $groupsMapping): void {
+        if ($groupsMapping->isEnabled()) {
+            $groups = $this->getGroupsFromProvider($groupsMapping->getEndpoint());
+            $this->validateGroupsMapping($groups, $groupsMapping);
+        } else {
+            $this->logAuthenticationInfo("Groups Mapping disabled", []);
```
Comment (kduret) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]: message will be formatted like `"[Openid] [INFO] $message : " . json_encode($content)`
so `"[Openid] [INFO] Groups Mapping disabled : {}"`
maybe, logAuthenticationInfo should be updated to not log content when it's empty

**Review by dmyios — APPROVED** [review]

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 4]
```diff
@@ -863,13 +867,13 @@ private function logAuthenticationDebug(string $message, array $content): void
...
-     * @param array<string|int,string> $content
+     * @param array<string|int,string>|null $content
      */
-    private function logAuthenticationInfo(string $message, array $content): void
+    private function logAuthenticationInfo(string $message, ?array $content = null): void
     {
         $this->centreonLog->insertLog(
             CentreonUserLog::TYPE_LOGIN,
-            "[Openid] [INFO] $message : " . json_encode($content)
+            "[Openid] [INFO] $message : " . $content !== null ? json_encode($content) : ''
```
Comment (kduret) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
            "[Openid] [INFO] $message" . ($content !== null ? ' : ' . json_encode($content) : '')
```

**Review by kduret — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**